### PR TITLE
feat(terraform): include pods of kubernetes_deployment in kubernetes_pod checks (1/4)

### LINKED ISSUE #3690: Include Pods specefied via kubernetes_deployments in kubernetes_pod checks
**Describe the issue**

Pods can also be defined via `kubernetes_deployment`. All `kubernetes_pod` checks should also work with pods specefied via `kubernetes_deployment`

**Examples**
```ebnf
resource "kubernetes_deployment" "deployment" {
  metadata {
    name = "terraform-example"
    labels = {
      k8s-app = "nginx"
    }
  }

  spec {
    replicas = 3

    selector {
      match_labels = {
        k8s-app = "nginx"
      }
    }

    template {
      metadata {
        labels = {
          k8s-app = "nginx"
        }
      }

      spec {
        container {
          image = "nginx:1.7.9"
          name  = "example"

          port {
            container_port = 8080
          }
      }
   }
}
```

```ebnf
resource "kubernetes_pod" "pod" {
  metadata {
    name = "terraform-example"
  }

  spec {
    container {
       image = "nginx:1.7.9"
       name  = "example"

       port {
         container_port = 8080
       }
     }
  }
}
```

**Version (please complete the following information):**
 - Checkov Version 2.1.270

### PR DESCRIPTION
**Pull request by ugrave**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

This PR includes pods defined via `kubernetes_deployment` in pod checks . PR 1 of 4.

Fixes #3690

## New/Edited policies (Delete if not relevant)
* CKV_K8S_20
   Add resource: kubernetes_deployment, kubernetes_deployment_v1
* CKV_K8S_25
  Add resource: kubernetes_deployment, kubernetes_deployment_v1
* CKV_K8S_39
  Add resource: kubernetes_deployment, kubernetes_deployment_v1
* CKV_K8S_11
  Add resource: kubernetes_deployment, kubernetes_deployment_v1
* CKV_K8S_10
  Add resource: kubernetes_deployment, kubernetes_deployment_v1
* CKV_K8S_30
  Add resource: kubernetes_deployment, kubernetes_deployment_v1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/checks/resource/kubernetes/AllowPrivilegeEscalation.py [hunk 1]
```diff
@@ -24,6 +25,14 @@ def scan_resource_conf(self, conf) -> CheckResult:
             return CheckResult.UNKNOWN
         spec = spec_list[0]
         if spec:
+            evaluated_keys_path = "spec"
+
+            if spec.get("template") and isinstance(spec.get("template"), list):
+                template = spec.get("template")[0]
+                if template.get("spec") and isinstance(template.get("spec"), list):
+                    spec = template.get("spec")[0]
+                evaluated_keys_path = f'{evaluated_keys_path}/[0]/template/[0]/spec'
```
Comment (gruebel) [inline checkov/terraform/checks/resource/kubernetes/AllowPrivilegeEscalation.py:34]:
```suggestion
            template = spec.get("template")
            if template and isinstance(template, list):
                template_spec = template[0].get("spec")
                if template_spec and isinstance(template_spec, list):
                    spec = template_spec[0]
                evaluated_keys_path = f'{evaluated_keys_path}/[0]/template/[0]/spec'
```
this way you don't access the `dict` multiple times to retrieve the same value. Same goes for the other checks you updated.

File: tests/terraform/checks/resource/kubernetes/test_CPULimits.py [hunk 2]
```diff
@@ -31,15 +33,23 @@ def test(self):
...
-
-        self.assertEqual(summary["passed"], 1 * 2)
-        self.assertEqual(summary["failed"], 4 * 2)
-        self.assertEqual(summary["skipped"], 0)
-        self.assertEqual(summary["parsing_errors"], 0)
+        #
+        # self.assertEqual(summary["passed"], 2 * 2)
+        # self.assertEqual(summary["failed"], 8 * 2)
+        # self.assertEqual(summary["skipped"], 0)
+        # self.assertEqual(summary["parsing_errors"], 0)
```
Comment (gruebel) [inline tests/terraform/checks/resource/kubernetes/test_CPULimits.py:52]: probably uncommenting it again would good 😄

**Review by gruebel — COMMENTED** [review]
great work again. I have specific change suggestion, which you probably need to add to all the adjusted checks, also in the other PRs.

File: tests/terraform/checks/resource/kubernetes/test_CPULimits.py (hunk 2 above)
Comment (ugrave) [reply tests/terraform/checks/resource/kubernetes/test_CPULimits.py]: Ups ;)

File: checkov/terraform/checks/resource/kubernetes/AllowPrivilegeEscalation.py (hunk 1 above)
Comment (ugrave) [reply checkov/terraform/checks/resource/kubernetes/AllowPrivilegeEscalation.py]: Changed this also in the other PRs-

**Review by gruebel — APPROVED** [review]
nice, thanks 🍹

File: checkov/terraform/checks/resource/kubernetes/AllowPrivilegeEscalation.py [hunk 3]
```diff
@@ -24,6 +25,16 @@ def scan_resource_conf(self, conf) -> CheckResult:
             return CheckResult.UNKNOWN
         spec = spec_list[0]
         if spec:
+            evaluated_keys_path = "spec"
+
+            template = spec.get("template")
+            if template and isinstance(template, list):
+                template = template[0]
+                template_spec = template.get("spec")
+                if template_spec and isinstance(template_spec, list):
```
Comment (tronxd) [inline checkov/terraform/checks/resource/kubernetes/AllowPrivilegeEscalation.py:34]:
```suggestion
                if template_spec and isinstance(template_spec, list) and len(template_spec) == 1:
```

File: checkov/terraform/checks/resource/kubernetes/CPULimits.py [hunk 4]
```diff
@@ -19,6 +20,15 @@ def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
             self.evaluated_keys = [""]
             return CheckResult.FAILED
         spec = conf['spec'][0]
+        evaluated_keys_path = "spec"
+
+        template = spec.get("template")
+        if template and isinstance(template, list):
```
Comment (tronxd) [inline checkov/terraform/checks/resource/kubernetes/CPULimits.py:26]:
```suggestion
        if template and isinstance(template, list) and len(template) == 1:
```

File: checkov/terraform/checks/resource/kubernetes/CPURequests.py [hunk 5]
```diff
@@ -15,6 +16,15 @@ def scan_resource_conf(self, conf) -> CheckResult:
             self.evaluated_keys = [""]
             return CheckResult.FAILED
         spec = conf['spec'][0]
+        evaluated_keys_path = "spec"
+
+        template = spec.get("template")
+        if template and isinstance(template, list):
```
Comment (tronxd) [inline checkov/terraform/checks/resource/kubernetes/CPURequests.py:22]:
```suggestion
        if template and isinstance(template, list) and len(template) == 1:
```

File: checkov/terraform/checks/resource/kubernetes/CPURequests.py [hunk 6]
```diff
@@ -15,6 +16,15 @@ def scan_resource_conf(self, conf) -> CheckResult:
...
+            template = template[0]
+            template_spec = template.get("spec")
+            if template_spec and isinstance(template_spec, list):
```
Comment (tronxd) [inline checkov/terraform/checks/resource/kubernetes/CPURequests.py:25]:
```suggestion
            if template_spec and isinstance(template_spec, list) and len(template_spec) == 1:
```

File: checkov/terraform/checks/resource/kubernetes/ContainerSecurityContext.py [hunk 7]
```diff
@@ -11,20 +11,31 @@ def __init__(self):
...
+                               'kubernetes_deployment', 'kubernetes_deployment_v1']
         categories = [CheckCategories.GENERAL_SECURITY]
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
     def scan_resource_conf(self, conf) -> CheckResult:
         spec = conf.get('spec', [None])[0]
+        evaluated_keys_path = "spec"
+
+        template = spec.get("template")
+        if template and isinstance(template, list):
```
Comment (tronxd) [inline checkov/terraform/checks/resource/kubernetes/ContainerSecurityContext.py:24]:
```suggestion
        if template and isinstance(template, list)  and len(template) == 1:
```

File: checkov/terraform/checks/resource/kubernetes/ContainerSecurityContext.py [hunk 8]
```diff
@@ -11,20 +11,31 @@ def __init__(self):
...
+            template = template[0]
+            template_spec = template.get("spec")
+            if template_spec and isinstance(template_spec, list):
```
Comment (tronxd) [inline checkov/terraform/checks/resource/kubernetes/ContainerSecurityContext.py:27]:
```suggestion
            if template_spec and isinstance(template_spec, list) and  and len(template_spec) == 1:
```

**Review by tronxd — APPROVED** [review]

**Pull request merged**